src/Core/Addon/Module/ModuleRepository.php

```diff
@@ -205,32 +205,30 @@ public function getFilteredList(AddonListFilter $filter, $skip_main_class_attrib
             // Part Two : Remove module not installed if specified
             if ($filter->status != AddonListFilterStatus::ALL) {
                 if ($module->database->get('installed') == 1
-                    && ($filter->hasStatus(AddonListFilterStatus::UNINSTALLED)
-                        || !$filter->hasStatus(AddonListFilterStatus::INSTALLED))) {
+                    && $filter->hasStatus(AddonListFilterStatus::UNINSTALLED)
+                    && !$filter->hasStatus(AddonListFilterStatus::INSTALLED)) {
                     unset($modules[$key]);
 
                     continue;
                 }
 
                 if ($module->database->get('installed') == 0
-                    && (!$filter->hasStatus(AddonListFilterStatus::UNINSTALLED)
-                        || $filter->hasStatus(AddonListFilterStatus::INSTALLED))) {
+                    && !$filter->hasStatus(AddonListFilterStatus::UNINSTALLED)
+                    && $filter->hasStatus(AddonListFilterStatus::INSTALLED)) {
```

Comment:
What is the relationship between this change and the PR scope?

Reply:
I had trouble making the test for this one work in process isolation And I went too deep, and the tests themselves didn't make sense 😅 And the code itself didn't make sense either!! I got lost somewhere in the middle of my own insanity...

It's a bit out of scope though, I can try and revert this to see if it goes fine now that the rest is cleaner and more stable if you want?

Reply:
Actually the previous tests mostly did nothing, half of them didn't perform any assertions because the filtered list didn't return anything Now there's only one left which does no assertions because it filters modules from addons and I couldn't find a way to mock this one since we are getting rid of Addon anyway

Reply:
Not sure it's a good idea to touch this part of code as it concerns module's bitwise :p 
I already experiment issues by editing it :eyes: